### PR TITLE
added `login` command

### PR DESCRIPTION
**Pull request by Taimoor-12**
The user can enter their API key using `mslm login` command either through a flag, an argument or by typing it after a CLI prompt. The key is then stored in the config.

### CODE REVIEW

File: mslm/cmd_login.go [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+		"--help":     predict.Nothing,
+	},
+}
+
+func printHelpLogin() {
+	fmt.Printf(
+		`Usage: %s login [<opts>]
+
+Description:
+  Enter the API key associated with Mslm account to authenticate
```
Comment (UmanShahzad) [inline mslm/cmd_login.go:30]:
```suggestion
  Authenticate with an Mslm account using an API key.
```

File: mslm/cmd_login.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+  in order to make API calls.
```
Comment (UmanShahzad) [inline mslm/cmd_login.go:31]: Rm this line

File: mslm/cmd_login.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+
+  The key will be saved and automatically used when an API command
+  is ran.
```
Comment (UmanShahzad) [inline mslm/cmd_login.go:34]: Remove these 2 lines above. Ensure no unnecessary newlines after that

File: mslm/cmd_login.go [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+		fmt.Println("please enter a key")
+	}
+
+	return key, nil
+}
+
+func checkValidity(key string) (bool, error) {
+	fmt.Println("checking key...")
+	// make API req for true key validity.
+	res, err := http.Get("https://mslm.io/api/v1/acct/apikey/zapier_check/0aa76ca0-bed2-4d07-815c-381c1b4c4084?apikey=" + key)
```
Comment (UmanShahzad) [inline mslm/cmd_login.go:143]: No, no, no. What does zapier check have to do with this? Remove this please, this is supposed to be a private API

Comment (UmanShahzad) [reply mslm/cmd_login.go:143]: You can remove the validity check. We only kept it around for zapier because they forced us to hav eit

Comment (Taimoor-12) [reply mslm/cmd_login.go:143]: Upon checking again, I found out this endpoint url is in `pub_api` and the function behind it basically checks for the existence/validity of an Mcloud Acct API key in the DB. Should I still remove this validity check in the command or should the URL path be changed?

Comment (UmanShahzad) [reply mslm/cmd_login.go:139]: Dont use that API here, it's meant to be a 'private' public API, but now we've revealed it in open source github comments 😄

File: mslm/cmd_login.go [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+	if key == "" {
+		newKey, err := enterKey(key)
+		if err != nil {
+			return fmt.Errorf(err.Error())
+		}
+
+		key = newKey
+	}
+
+	if err := UpdateConfigFieldAndSave("ApiKey", key); err != nil {
```
[the remark this answers is not in the document]
Comment (Taimoor-12) [reply mslm/cmd_login.go:88]: Did this change in the logout PR

**Review by UmanShahzad — APPROVED** [review]

**Pull request merged**